### DISCUSSION TITLE
Reading data from device with specific interval

### DISCUSSION THREAD

**Discussion by majidalikhan8baader** [edgexfoundry/community#52] [Q&A] [answered]
Normally device service listens to the device and whenever an event is available, it pushes the event on the message bus and we can read that in the application service via default pipelines. But, what if I want the device service to listen to the device only after a specific interval, how does one do it?  Such as I want to read from the device at 1-minute intervals and read the data that is available at that time.

I haven't been able to find any interval configurations in the device services or application service. Any hints?

**Comment by jpwhitemn** [comment, marked as answer]
@majidalikhan8baader - use autoevents to trigger the read of the device resources on the frequency you want.  See https://docs.edgexfoundry.org/2.3/design/legacy-requirements/device-service/#autoevents.  You can see an example of scheduled autoevents with device resources on the Virtual device service in the device configuration here:  https://github.com/edgexfoundry/device-virtual-go/blob/main/cmd/res/devices/devices.toml

**Reply by majidalikhan8baader** [reply]
@jpwhitemn thanks for the links
